### ISSUE TITLE
RetroArch Integration

### ISSUE THREAD

**Issue by RelativeProgramming** [Mixaill/awesome-gog-galaxy#37]
It would be nice, to have an integration of RetroArch.
This could be a comfortable alternative to the single emulator integrations, that already exist.

**Comment by UncleGoogle** [comment]
Wow, that is huge. Adding for reference: https://www.libretro.com/index.php/home-2/

There is also http://retroachievements.org/ that can be integrated but I'm not sure if within the same plugin.

**Comment by borger** [comment]
Agree! Galaxy requires one plugin per platform with a unique id. I believe there could be a few plugins using the same structure on the backend. Similar to what Dolphin has in this repo's list. Love the idea of integrating with RetroAchievements!

**Comment by JTNDev** [comment]
I was thinking of making the Dolphin plugins work with Libretro, but of
course I wouldn't be able to get RetroAchievements to work because it
doesn't support GCN.

**Comment by borger** [comment]
It doesn’t? I though the Dolphin core on Retroarch did support RetroAchievements, not sure if it does... just something to keep in mind..

**Comment by JTNDev** [comment]
I don't think it does, it shows all the games here and there doesn't seem
to be a GameCube or Wii section yet:
http://retroachievements.org/gameList.php

**Comment by AndrewDWhite** [comment]
I created a prototype to generically implement this. If you want to try it out, the gitlab project is here. 
https://github.com/AndrewDWhite/GalaxyGenericImporterPlugin
I need to clean it up and add documentation, but let me know if the proof of concept is along the lines of what you were looking for.

**Comment by UncleGoogle** [comment]
@AndrewDWhite nice.

What are advantages over having multiple platforms in currnely listed retroarch plugin https://github.com/jshackles/RetroGOG?

For sure size, and no need to code generator - makes changes in one place :)

But drawback is that in Galaxy all those will appear as "Test" - only one platform. Maybe Galaxy should add "RetroArch" platform id

**Comment by AndrewDWhite** [comment]
I thought about implementing it the same way, where it is spread around the differing platforms and there are two advantages for this. Galaxy is missing a bunch of platforms so I placed it in the same platform to allow all platform to be integrated even if they are not on the list. The other advantage is that some advanced searching may be easier with tags if you want an 'or' function rather than an 'and' function.

It also makes sense to organize it the same way allowing the user to auto populate even more tags, for example RetroArch, and place any unavailable systems in testing until the platforms are integrated.
